org.xml.sax.SAXParseException: XML document structures must start and end within the same entity

# Task Description

We are occasionally seeing the following random test failure:

```
javax.xml.bind.UnmarshalException
at org.carlspring.strongbox.storage.repository.RepositoryTest.testMarshallAndUnmarshallStrongboxConfiguration(RepositoryTest.java:113)
Caused by: org.xml.sax.SAXParseException: XML document structures must start and end within the same entity.
at org.carlspring.strongbox.storage.repository.RepositoryTest.testMarshallAndUnmarshallStrongboxConfiguration(RepositoryTest.java:113)
Standard Output
21:49:43.642 21-03-2019 | INFO | kJoinPool-1-worker-0 | s.t.s.r.TestRepositoryManagementApplicationContext | Test Repository [storage0/r1] locked.
21:49:43.642 21-03-2019 | INFO | kJoinPool-1-worker-0 | s.t.s.r.TestRepositoryManagementApplicationContext | Test Repository [storage0/r2] locked.
```

We need to investigate and fix this.

# Related Tasks

* This test has been disabled in the `master` until #965 has been fixed. @ptirador , @fuss86 : If no one has fixed it by then, (or if you haven't fixed in on your branch), could you please pick this up?

* This task may somehow be related to #1105 (unconfirmed).

# Acceptance Test

* [ ] Building the code and running `mvn spring-boot:run` in the `strongbox-web-core` still starts up the application correctly.
* [ ] Building the code and running the `strongbox-distribution` from a `zip` or `tar.gz` still works.
* [ ] The tests in the [`strongbox-web-integration-tests`](https://github.com/strongbox/strongbox-web-integration-tests/) still run properly.

# Help

* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @carlspring
  * @sbespalov
  * @fuss86

--- a/strongbox-storage/strongbox-storage-api/src/test/java/org/carlspring/strongbox/storage/repository/RepositoryTest.java
+++ b/strongbox-storage/strongbox-storage-api/src/test/java/org/carlspring/strongbox/storage/repository/RepositoryTest.java
@@ -8,7 +8,6 @@ import org.carlspring.strongbox.xml.parsers.GenericParser;
 
 import javax.xml.bind.JAXBException;
 import java.io.ByteArrayInputStream;
-import java.io.File;
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
@@ -107,10 +106,8 @@ public class RepositoryTest
     public void testMarshallAndUnmarshallStrongboxConfiguration()
             throws JAXBException, IOException
     {
-        File file = new File("target/strongbox/etc/conf/strongbox.xml");
-
         GenericParser<MutableConfiguration> parser = new GenericParser<>(MutableConfiguration.class);
-        MutableConfiguration configuration = parser.parse(file.toURI().toURL());
+        MutableConfiguration configuration = parser.parse(this.getClass().getResourceAsStream("/etc/conf/strongbox.xml"));
 
         MutableStorage storage = configuration.getStorage("storage0");
 

--- a/strongbox-storage/strongbox-storage-api/src/test/java/org/carlspring/strongbox/testing/artifact/TestArtifactContext.java
+++ b/strongbox-storage/strongbox-storage-api/src/test/java/org/carlspring/strongbox/testing/artifact/TestArtifactContext.java
@@ -1,6 +1,7 @@
 package org.carlspring.strongbox.testing.artifact;
 
 import java.io.IOException;
+import java.io.InputStream;
 import java.net.URI;
 import java.nio.file.Files;
 import java.nio.file.Path;
@@ -82,10 +83,9 @@ public class TestArtifactContext implements AutoCloseable
                       .values()
                       .stream()
                       .forEach(p -> {
-                          try
+                          try(InputStream is = Files.newInputStream(artifactPathLocal.resolveSibling(p.getFileName())))
                           {
-                              artifactManagementService.store(p,
-                                                              Files.newInputStream(artifactPathLocal.resolveSibling(p.getFileName())));
+                              artifactManagementService.store(p,is);
                           }
                           catch (IOException e)
                           {

